### PR TITLE
project skeleton

### PR DESCRIPTION
**Pull request by omer182**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+![npm](https://img.shields.io/npm/v/axios-time)
+
+
+# axios-timing-plugin
```
Comment (kobik) [inline README.md]: should be `axios-time`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Axios plugin to measure the actual time it takes for a request to complete
+
+## Installation
+You can install this plugin using **NPM**
+```bash
+> npm i axios-timing-plugin
```
Comment (kobik) [inline README.md]: should be `axios-time`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+```
+
+## Basic Usage
+The example below will add timing data to the request-response cycle.
+```js
+const axios = require('axios');
+const axiosTimingPlugin = require('src/index');
```
Comment (kobik) [inline README.md]: should be `axios-time`

File: package.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+    "name": "axios-timing-plugin",
```
Comment (kobik) [inline package.json]: should be `axios-time`

File: package.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    "version": "1.0.0",
+    "description": "Axios plugin to measure the actual time it takes for a request to complete",
+    "main": "src/index.js",
+    "scripts": {
+        "test": "./node_modules/.bin/_mocha ./tests/test"
```
Comment (kobik) [inline package.json]: `./node_modules/.bin/_mocha` should be `$(npm bin)/mocha`

File: package.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    },
+    "author": "PayU",
+    "license": "Apache-2.0",
+    "files": [
+        "src"
+    ],
+    "dependencies": {
+        "chai": "^4.2.0"
```
Comment (kobik) [inline package.json]: dev-dependency

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+module.exports = function (instance) {
+    instance.interceptors.request.use(function (request) {
+        request.timingStart = Date.now();
+        return request;
+    });
+
+    instance.interceptors.response.use(function (response) {
```
Comment (kobik) [inline src/index.js:9]: timings should be added also when axios throws an error

File: tests/test.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('testing axios-time package', () => {
+    it('should have described timing properties correctly', async () => {
+        try {
+            const response = await axios.get('http://ynet.co.il');
+            expect(response.timings).to.have.property('timingEnd').that.is.a('number');
+            expect(response.timings).to.have.property('timingStart').that.is.a('number');
+            expect(response.timings).to.have.property('elapsedTime').that.is.a('number');
+        } catch(err) {
+            throw(err);
```
Comment (kobik) [inline tests/test.js]: redundant try-catch if throwing the only thing that is done

**Review by kobik — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (omer182) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (omer182) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (omer182) [reply README.md]: fixed

File: package.json (hunk 4 above)
Comment (omer182) [reply package.json]: fixed

File: package.json (hunk 5 above)
Comment (omer182) [reply package.json]: fixed

File: package.json (hunk 6 above)
Comment (omer182) [reply package.json]: fixed

File: tests/test.js (hunk 8 above)
Comment (omer182) [reply tests/test.js]: fixed

**Review by kobik — APPROVED** [review]

**Pull request merged**